lib/ansible/module_utils/tacp.py

```diff
@@ -246,12 +248,20 @@ def wait_for_action_to_complete(action_uuid, api_client):
         api_instance = tacp.ActionsApi(api_client)
         action_incomplete = True
 
-        while action_incomplete:
+        timeout = 30
+        time_spent = 0
+
+        while action_incomplete and time_spent < timeout:
```

Comment:
i don't know if that would do what I need since I know the request will return something, but I don't know if the task will ever succeed in a timely fashion. My intent here is to keep checking until the status changes to Completed and just give up after 30 seconds. I don't think your suggestion would do that, or do I misunderstand?

Reply:
Right, got it!